cmd/soroban-rpc/lib/preflight/src/fees.rs

```diff
@@ -436,3 +445,41 @@ pub(crate) fn compute_restore_footprint_transaction_data_and_min_fee(
         bucket_list_size,
     )
 }
+
+fn compute_restore_footprint_rent_changes(
+    footprint: &LedgerFootprint,
+    ledger_storage: &LedgerStorage,
+    min_persistent_entry_expiration: u32,
+    current_ledger_seq: u32,
+) -> Result<Vec<LedgerEntryRentChange>> {
+    let mut rent_changes: Vec<LedgerEntryRentChange> = Vec::new();
+    for key in footprint.read_write.as_vec() {
+        let unmodified_entry = ledger_storage
+            .get(key, true)
+            .with_context(|| format!("cannot get ledger entry with key {key:?}"))?;
+        let size = (key.to_xdr()?.len() + unmodified_entry.to_xdr()?.len()) as u32;
+        let expirable_entry: Box<dyn ExpirableLedgerEntry> =
+            (&unmodified_entry).try_into().map_err(|e: String| {
+                Error::msg(e.clone()).context("incorrect ledger entry type in footprint")
```

Comment:
could you change this to be more specific ? 
`The ledger key specified in the footprint read-write section was an expirable ledger entry; these entry types cannot be restored directly`

Reply:
The parent error will tell you that (i.e. `e.clone()`). See `state_expiration.rs`